### PR TITLE
Convert post to get 153

### LINKED ISSUE #153: Convert cacheable API requests from POST to GET
Currently, our `courseinfo`, `filtercoursecodes`, and `recommendedsequence` API endpoints are using POST. These requests respond with the same JSON object each time they are called with equivalent body params, and they don't produce any side effects on the backend since they only do reads on the db. This means that we can safely benefit from [http caching](https://developers.google.com/web/fundamentals/performance/optimizing-content-efficiency/http-caching) . This would be an easy change and result in a faster frontend and a reduced load on the server every time a request is repeated by the same client.

As such, these three endpoints should be changed to use GET instead of POST, and more importantly the backend should send the appropriate headers to allow caching. This will require code changes on the frontend and backend.

### PR DESCRIPTION
**Pull request by PeterGhimself**
closes #153 

## Summary
Changed `recommendedsequences`, `courseinfo` and `filtercourses` requests to use GET instead of POST on both front-end and back-end for better cache optimisation.

### CODE REVIEW

**Review by Davidster — APPROVED** [review]
`bilbo swaggins`

**Review by Davidster — CHANGES_REQUESTED** [review]
image

**Comment by Davidster** [comment]
Jokes aside... `src/mainPage/searchBox.test.js` is currently failing. Use command `npm run test -- --verbose` to run unit tests

**Review by Davidster — CHANGES_REQUESTED** [review]

File: frontend/src/mainPage/searchBox.test.js [hunk 1]
```diff
@@ -64,17 +64,16 @@ describe("class functions", () => {
             expect(searchBoxClass.setState.mock.calls[0][0]).toEqual({ isFiltering: true });
         });
 
-        test("should perform a POST request at api/filtercoursecodes", () => {
+        test("should perform a GET request at api/filtercoursecodes?filter=com", () => {
```
[the remark this answers is not in the document]
Comment (PeterGhimself) [reply frontend/src/mainPage/searchBox.test.js]: done, thanks

File: frontend/src/mainPage/searchBox.test.js [hunk 2]
```diff
@@ -64,17 +64,16 @@ describe("class functions", () => {
...
 
             let mockCalls = $.ajax.mock.calls;
             expect(mockCalls.length).toBe(1);
 
             let ajaxArgs = mockCalls[0][0];
-            expect(ajaxArgs.type).toBe("POST");
-            expect(ajaxArgs.url).toBe("api/filtercoursecodes");
-            expect(ajaxArgs.data).toBe(JSON.stringify({ filter: searchQuery }));
+            expect(ajaxArgs.type).toBe("GET");
+            expect(ajaxArgs.url).toBe("api/filtercoursecodes?filter=com");
```
[the remark this answers is not in the document]
Comment (PeterGhimself) [reply frontend/src/mainPage/searchBox.test.js]: done, thanks

**Review by Davidster — APPROVED** [review]

**Pull request merged**